mcache/tests/test_check.py

```diff
@@ -134,14 +135,56 @@ def memcached():
 
     yield
 
-    subprocess.check_call(["docker-compose", "-f", os.path.join(HERE, 'docker-compose.yaml'), "down"])
+    subprocess.check_call(["docker-compose", "-f", docker_compose_file, "down"])
+
+
+@pytest.fixture(scope="session")
+def memcached_socket():
+    """
+    Start a standalone Memcached server.
+    """
+    env = os.environ
+    env['PWD'] = HERE
+    env['DOCKER_SOCKET_DIR'] = DOCKER_SOCKET_DIR
+    env['DOCKER_SOCKET_PATH'] = DOCKER_SOCKET_PATH
+    env['UNIXSOCKET_DIR'] = UNIXSOCKET_DIR
+
+    os.chmod(UNIXSOCKET_DIR, 00777)
+
+    docker_compose_file = os.path.join(HERE, 'compose', 'docker-compose.yaml')
+    subprocess.check_call(["docker-compose", "-f", docker_compose_file, "up", "-d", "memcached_socket"], env=env)
+
+    attempts = 0
+    while True:
+        if attempts > 10:
+            raise Exception("Memcached boot timed out!")
+
+        mc = bmemcached.Client(UNIXSOCKET_PATH, USERNAME, PASSWORD)
+        try:
+            mc.set("foo", "bar")
+        except MemcachedException:
+            attempts += 1
+            time.sleep(1)
+        else:
+            mc.delete("foo")
+            mc.disconnect_all()
+            break
+
+    yield
+
+    subprocess.check_call(["docker-compose", "-f", docker_compose_file, "down"])
```

Comment:
Let's remove the temp directory we created with `shutil.rmtree` as @ofek mentioned